aws-lc-fips-sys/builder/cmake_builder.rs

```diff
@@ -0,0 +1,196 @@
+}
+
+fn test_go_command() -> bool {
+    test_command("go".as_ref(), &["version".as_ref()])
+}
+
+fn find_cmake_command() -> Option<&'static OsStr> {
+    if test_command("cmake3".as_ref(), &["--version".as_ref()]) {
+        Some("cmake3".as_ref())
+    } else if test_command("cmake".as_ref(), &["--version".as_ref()]) {
+        Some("cmake".as_ref())
+    } else {
+        None
+    }
+}
+
+fn get_platform_output_path() -> PathBuf {
+    PathBuf::new()
+}
+
+impl CmakeBuilder {
+    pub(crate) fn new(
+        manifest_dir: PathBuf,
+        out_dir: PathBuf,
+        build_prefix: Option<String>,
+        output_lib_type: OutputLibType,
+    ) -> Self {
+        Self {
+            manifest_dir,
+            out_dir,
+            build_prefix,
+            output_lib_type,
+        }
+    }
+
+    fn artifact_output_dir(&self) -> PathBuf {
+        self.out_dir
+            .join("build")
+            .join("artifacts")
+            .join(get_platform_output_path())
+    }
+
+    fn get_cmake_config(&self) -> cmake::Config {
+        cmake::Config::new(&self.manifest_dir)
+    }
+
+    fn prepare_cmake_build(&self) -> cmake::Config {
+        let mut cmake_cfg = self.get_cmake_config();
+
+        if ["powerpc64", "powerpc"]
+            .iter()
+            .any(|arch| target_arch().eq_ignore_ascii_case(arch))
+        {
+            cmake_cfg.define("ENABLE_EXPERIMENTAL_BIG_ENDIAN_SUPPORT", "1");
+        }
+
+        if OutputLibType::default() == OutputLibType::Dynamic {
+            cmake_cfg.define("BUILD_SHARED_LIBS", "1");
+        } else {
+            cmake_cfg.define("BUILD_SHARED_LIBS", "0");
+        }
+
+        let opt_level = env::var("OPT_LEVEL").unwrap_or_else(|_| "0".to_string());
+        if opt_level.ne("0") {
+            if opt_level.eq("1") || opt_level.eq("2") {
+                cmake_cfg.define("CMAKE_BUILD_TYPE", "relwithdebinfo");
+            } else {
+                cmake_cfg.define("CMAKE_BUILD_TYPE", "release");
+            }
+        } else if target_os() != "windows" {
+            // The Windows/FIPS build rejects "debug" profile
+            // https://github.com/aws/aws-lc/blob/main/CMakeLists.txt#L656
+            cmake_cfg.define("CMAKE_BUILD_TYPE", "debug");
+        }
+
+        if let Some(prefix) = &self.build_prefix {
+            cmake_cfg.define("BORINGSSL_PREFIX", format!("{prefix}_"));
+            let include_path = self.manifest_dir.join("generated-include");
+            cmake_cfg.define(
+                "BORINGSSL_PREFIX_HEADERS",
+                include_path.display().to_string(),
+            );
+        }
+
+        // Build flags that minimize our crate size.
+        cmake_cfg.define("BUILD_TESTING", "OFF");
+        if cfg!(feature = "ssl") {
+            cmake_cfg.define("BUILD_LIBSSL", "ON");
+        } else {
+            cmake_cfg.define("BUILD_LIBSSL", "OFF");
+        }
+
+        if target_vendor() == "apple" {
+            if target_os().trim() == "ios" {
```

Comment:
no need to `.trim()` on other calls to `target_os()` (e.g. L88)?

i suppose similar question applies to non-fips